base/int.jl

```diff
@@ -482,6 +482,33 @@ julia> trailing_ones(3)
 """
 trailing_ones(x::Integer) = trailing_zeros(~x)
 
+"""
+    top_set_bit(x::Integer) -> Integer
+
+The number of bits in `x`'s binary representation, excluding leading zeros.
+
+Equivalently, the position of the most significant set bit in `x`'s binary
+representation, measured from the least significant side.
+
+Negative `x` are only supported when `x::BitSigned`.
+
+`top_set_bit` is internal and will eventually be replaced by constant propagation.
+
```

Comment:
```suggestion
```
The previous plan was to remove it once `Base.ndigits0z(x,2)` became equally as efficient due to constant propagation, but if folks think the end game may be exporting rather than removal (:-1: for increased code size) then maybe we should cut this line in the docstring.

Reply:
I think cut the line.

Reply:
`Base.ndigits0z(x,2)` and `top_set_bit(x)` do return the same result for all `x::BitUnsigned`